src/include/simeng/OS/Process.hh

```diff
@@ -110,12 +136,17 @@ class Process {
 
   /** Get the process' TID. */
   uint64_t getTID() const { return TID_; }
+  /** Method which handles a page fault. */
+  uint64_t handlePageFault(uint64_t vaddr, SendToMemory send);
+
+  /** Method which handles virtual address translation. */
+  uint64_t translate(uint64_t vaddr) { return pageTable_->translate(vaddr); }
 
-  /** The virtual file descriptor mapping table. */
-  std::vector<int64_t> fileDescriptorTable_;
+  /** Method which return reference to page table shared_ptr. */
+  std::shared_ptr<PageTable>& getPageTable() { return pageTable_; }
 
-  /** Set of deallocated virtual file descriptors available for reuse. */
-  std::set<int64_t> freeFileDescriptors_;
+  /** Shared pointer to FileDescArray class.*/
+  std::unique_ptr<FileDescArray> fdArray_;
```

Comment:
This was one of the action points brought up on the day we had our initial review, I had made it into a shared_ptr/pointer (don't recall properly) such that it could be shared easily when implementing the clone syscall (as fds are shared between cloned processes), however it was then decided to be changed into a unique_ptr
![image](https://user-images.githubusercontent.com/62028695/219714213-24b5de57-37ae-41a2-a88d-49b2d7a34391.png)

Reply:
ok noted. Please update comment to match member variable's type

Reply:
done